src/partielspy/plugin.py

```diff
@@ -0,0 +1,23 @@
+from lxml import etree
+
+from .plugin_key import PluginKey
+
+
+class Plugin:
+    def __init__(self, key: PluginKey = None):
+        if key is not None and not isinstance(key, PluginKey):
```

Comment:
Use if key is None -> self.__key = PluginKey(), else if not PluginKey  -> raise, else self.__key = key